File: 5_private_attribute.py
class A:
    __value = 0 #Private Attribute
    name = '' #Public Attribute
    
    def display(self):
        print("This is class")
        
    # def setter(self,val):
    #     self.__value = val
    
    @property   
    def get_value(self):
        return self.__value
    
    @get_value.setter
    def set_value(self,val):
        self.__value = val
        
    
obj = A()

obj.display()

# obj.value = 100
# print(obj._A__value)
obj.name = 'Object'
# obj.setter(40)

# print(obj.getter())
# print(obj.value)

obj.set_value = 89

print(obj.name)
print(obj.get_value)

File: 5_private_attribute.cpp
#include<bits/stdc++.h>

using namespace std;
#define fast ios_base::sync_with_stdio(0); cin.tie(NULL);
#define ll long long

class A
{
    int value; //Private Attribute

public:
    string str; //Public Attribute
    void display()
    {
        cout<<"This is class"<<endl;
    }

    void setter(int v)
    {
        value = v;
    }

    void getter()
    {
        cout<<value<<endl;
    }
};

void solve()
{
    A obj;

    obj.setter(100);
    obj.str = "Object";

    obj.display();

    obj.getter();
    cout<<obj.str<<endl;
}

int main()
{
    fast;
    solve();
    return 0;
}